### PR TITLE
SDL_image: exactly link with libpng and other recommended libraries

### PR DESCRIPTION
**Pull request by s172262**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

---

libSDL_image.dylib in  `sdl_image-1.2.12_2.el_capitan.bottle.tar.gz` was not linked libpng and other recommended libraries. There are two reasons.

First one is that --enable-feature-shared args work as inverse. See https://bugzilla.libsdl.org/show_bug.cgi?id=3308
The upper example in bug report page is libSDL_image.dylib in  `sdl_image-1.2.12_2.el_capitan.bottle.tar.gz`.
This revision's one links with depended libraries and zlib (the lower example). So we may set `--enable-feature-shared=no`.  But there was no shared object for libpng yet.

Second, to link with libpng and other recommended libraries needs to disable linkage with ImageIO framework. This is alternative.

Now we can get the shared symbols for libpng and others.

### CODE REVIEW

**Comment by s172262** [comment]
I have already built Lincity-ng and can play now with this revision. To pray Lincity-ng needs `brew unlink lincity-ng` and booting in Keg.

File: Formula/sdl_image.rb [hunk 1]
```diff
@@ -12,22 +12,26 @@ class SdlImage < Formula
     sha256 "84272135304e7c475e8000a90af113127858273a4449ce6566805f5b4c2cb476" => :mavericks
   end
 
+  option :universal
+
   depends_on "pkg-config" => :build
   depends_on "sdl"
-  depends_on "jpeg" => :recommended
```
Comment (dunn) [inline Formula/sdl_image.rb]: Did you mean to re-add this as a required dependency?

Comment (s172262) [inline Formula/sdl_image.rb]: `libtiff` requires `jpeg`. May I write explicitly?

Comment (dunn) [inline Formula/sdl_image.rb]: Yes, if `jpeg` is a direct dependency.

**Comment by s172262** [comment]
You can't wait any more.

**Pull request closed without merging**